### DISCUSSION TITLE
error trying to DeployMoodNftTest.t.sol

### DISCUSSION THREAD

**Discussion by GenevaRoc** [Cyfrin/foundry-full-course-cu#1055] [Q&A]
I keep getting an error trying to run the test for DeployMoodNftTest.t.sol.
Here is the error `Error: 
Compiler run failed:
Error (7576): Undeclared identifier.
  --> src/MoodNft.sol:40:14:
   |
40 |         if (!_isApprovedOrOwner(msg.sender, tokenId)) {
   |              ^^^^^^^^^^^^^^^^^^`

I traced the error to my moodNft contract in my src file. Don't know what could be the cause but, I followed every step duely

**Comment by Niferu** [comment]
Hi @GenevaRoc 

How can we help you if we do not even see your code? Maybe sharing repo would be good idea

**Comment by Amio13** [comment]
I think is _isApprovedOrOwner is no more in the Openzeppelin ERC721 Contracts .. replaced by _isAuthorized   read here https://github.com/OpenZeppelin/openzeppelin-contracts/blob/master/CHANGELOG.md

**Comment by Amio13** [comment]
Any solution for _exist at the TokenURI function in the tutorial...I know is removed and replaced by _ownerOf(tokenId) != address(0). 

But how do I effect it in the moodNft.sol at the TokenURI function
